### PR TITLE
Lock down more BLT version

### PR DESCRIPTION
**Pull request by white238**
I did my best here but this should get stakeholder's looking it over.

### CODE REVIEW

File: packages/camp/package.py [hunk 1]
```diff
@@ -216,7 +216,9 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cub", when="+cuda")
 
-    depends_on("blt")
+    depends_on("blt", type="build")
+    depends_on("blt@0.6.0:", type="build", when="@2023.06.99:")
```
Comment (rhornung67) [inline packages/camp/package.py:220]: Is the '99' correct here and below?

Comment (white238) [reply packages/camp/package.py:220]: So the `:` follows the Python list access pattern which is inclusive. So this is the convention some people use in Spack when you want to be above a version but not include it.

I am up for alternatives if you like, or I can go back when the new version shows up and fix this.

Comment (rhornung67) [reply packages/camp/package.py:220]: Thanks for the explanation. I can put up a PR in this project after the new releases come out.

**Comment by adrienbernede** [comment]
@white238 @rhornung67 We had 4 separate PRs related to February 2024 releases. I am gathering them in a dedicated PR and addressing the merge there.

**Pull request merged**